templates/sync-catalog-deployment.yaml

```diff
@@ -117,6 +123,18 @@ spec:
                 {{- if .Values.syncCatalog.addK8SNamespaceSuffix}}
                 -add-k8s-namespace-suffix \
                 {{- end}}
+                {{- if .Values.global.consulNamespacesEnabled }}
+                -enable-namespaces=true \
+                {{- if .Values.syncCatalog.consulNamespaces.consulDestinationNamespace }}
```

Comment:
What do you think about only setting the `-consul-namespace` flag *iff* mirrorK8S is `false`. This will have the same effect (mirroring overrides the flag) but if users look at the generated pod spec it will make a bit more sense as to what's going to occur.